File: backend/api/member/resume/models/personal_info.py
from .header import Header
from django.db import models


class PersonalInfo(models.Model):
    header = models.OneToOneField(
        Header, related_name="personal_info", on_delete=models.CASCADE, blank=True, null=True)
    full_name = models.CharField(max_length=200, blank=False)
    job_title = models.CharField(max_length=200, blank=False)
    email = models.EmailField(blank=True)
    phone_number = models.CharField(max_length=15, blank=True)
    address = models.CharField(max_length=255, blank=True)
    date_of_birth = models.DateField(blank=True, null=True)
    driving_license = models.CharField(max_length=50, blank=True)
    gender_pronoun = models.CharField(max_length=20, blank=True)
    marital_status = models.CharField(max_length=20, blank=True)
    nationality = models.CharField(max_length=50, blank=True)

File: frontend/src/lib/interfaces/resume/PersonalInfo.ts
export default interface PersonalInfo {
	id?: number | string;
	header?: number;
	full_name?: string;
	job_title?: string;
	email?: string;
	phone_number?: string;
	address?: string;
	date_of_birth?: string | Date;
	driving_license?: string;
	gender_pronoun?: string;
	marital_status?: string;
	nationality?: string;
}
